### ISSUE TITLE
Undefined symbols for architecture x86_64: _mono_runtime_set_pending_exception

### ISSUE THREAD

**Issue by marek-safar** [mono/mono#9211]
## Steps to Reproduce

1. Build XI

### Current Behavior

```
Undefined symbols for architecture x86_64:
  "_mono_runtime_set_pending_exception", referenced from:
      _xamarin_process_nsexception_using_mode in libxammac.a(runtime.static.x86_64.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Full log at https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/7420/testReport/Xamarin.MMP.Tests/TargetFrameworkDetectionTests/LongProfileStrings_SelectCorrectProfile/

### Expected Behavior

No error

## On which platforms did you notice this

[x] macOS
[ ] Linux
[ ] Windows

**Version Used**:

2018-04 integration

**Comment by rolfbjarne** [comment]
> Is it ok that the xam.mac build seems to link against the system mono ?

@vargaz yes, that's expected (the user can choose, and it's an option we test and support)
